azurelinuxagent/common/protocol/wire.py

```diff
@@ -961,7 +961,7 @@ def send_request_using_appropriate_channel(self, direct_func, host_func):
                              log_event=True)
                 raise
 
-        if not HostPluginProtocol.is_default_channel():
+        if ret and not HostPluginProtocol.is_default_channel():
```

Comment:
if this is a check against None (as opposed to a bool or an empty collection) let's be explicit: if ret is not None

Reply:
Sounds good.

Reply:
Actually, `azurelinuxagent.common.protocol.wire.WireClient.stream` is a lambda that can return False. I will make the check more explicit to include both None and False

Reply:
Decided that `if ret` looks better than `if ret is not None and if ret is not False`.